cmd/serverless-init/main.go

```diff
@@ -235,3 +239,23 @@ func setEnvWithoutOverride(envToSet map[string]string) {
 		}
 	}
 }
+
+func errorExitCode(err error) int {
+	// if err is of type errors.joinError then unwrap and find the first error with an exit code
+	if joinErr, ok := err.(interface{ Unwrap() []error }); ok {
+		for _, e := range joinErr.Unwrap() {
+			// if error is of type exec.ExitError then propagate the exit code
+			if exitError, ok := e.(*exec.ExitError); ok {
```

Comment:
You might consider using `errors.As` https://pkg.go.dev/errors#example-As

Reply:
Thank you! Was able to make this function much simpler while still achieving the same results

https://github.com/DataDog/datadog-agent/blob/7258622fcdc32568c4ebb0984a060ea831359963/cmd/serverless-init/main.go#L244-L253